Services/DataX.Utilities/DataX.Utilities.Web/RolesCheck.cs

```diff
@@ -35,9 +59,12 @@ public static void EnsureWriter(HttpRequest request)
             }
 
             var userrole = request.Headers[Constants.UserRolesHeader];
+            var userID = request.Headers[Constants.UserIdHeader];
+            ScenarioTestHelper();
+            
             Ensure.NotNull(userrole, "userrole");
 
-            if (!userrole.ToString().Contains(WriterRoleName))
+            if (!userrole.ToString().Contains(WriterRoleName) && !string.IsNullOrEmpty(userID) && !_ClientWhitelist.Contains(userID))
```

Comment:
is the userid never empty for human/non-human scenarios?

Reply:
Let's just say: If the userId is empty, the first check would have already failed. Since the role is associated with the userId. So, yes, it is never empty.